metadata-ingestion/src/datahub/ingestion/source/usage/usage_common.py

```diff
@@ -129,16 +129,17 @@ class GenericAggregatedDataset(Generic[ResourceType]):
 
     def add_read_entry(
         self,
-        user_email: str,
+        user_email: Optional[str],
         query: Optional[str],
         fields: List[str],
         user_email_pattern: AllowDenyPattern = AllowDenyPattern.allow_all(),
     ) -> None:
-        if not user_email_pattern.allowed(user_email):
+        if user_email and not user_email_pattern.allowed(user_email):
             return
 
         self.readCount += 1
-        self.userFreq[user_email] += 1
+        if user_email is not None:
+            self.userFreq[user_email] += 1
```

Comment:
This will cause the `userFreq` and total `queryCount` to potentially not match up. I think this is preferred over dropping the query entirely

Reply:
yup makes sense